### PR TITLE
Allow direct $apply syntax (closes #70)

### LINKED ISSUE #70: $apply syntax shortening
Now for a function application we need to use the syntax like that
```js
const mul = a => b => a * b
update(numbers, { one: { $apply: mul(2) }})
```

What do you think about shortening the syntax like that?
```js
update(numbers, { one: mul(2) })
```
Looks more declarative.

### PR DESCRIPTION
**Pull request by gnapse**

### CODE REVIEW

**Comment by gnapse** [comment]
A couple of things I'm not entirely sure how to address:

Since this moves away from the basic structure supported so far entirely based on commands, I wasn't sure how to address it in the README to explain the new supported functionality. I'll give it a shot in a separate commit, but I'm sure it may be seem off and suggestions are welcome.

Also related to this change moving away from all updates being based on commands: I was tempted to change the invariants at the top of the `update` function, to state that the `spec` had to be not only an object now, but either an object or a function. This complicated the invariant message, and I also noticed the message is being used verbatim in the tests to check if an invalid use of the `update` function indeed fails. Changing this message therefore has the implication that I have to update a few of otherwise unrelated tests. That's why I ended up deciding against it, and I put the check for the spec being a function right at the top, to get it off the way if it was not, and the rest of the functionality stays the same.

As usual, if any of this is not entirely satisfactory let me know.

File: test.js [hunk 1]
```diff
@@ -265,6 +265,28 @@ describe('update', function() {
     });
   });
 
+  describe('direct apply', function() {
+    var applier = function(node) {
+      return {v: node.v * 2};
+    };
+    it('applies', function() {
```
Comment (kolodny) [inline test.js:272]: Can you also add a test like this:

```js
it('applies', function() {
  var doubler = function(value) { return value * 2; };
  expect(update(2, doubler)).toEqual(4);
});
```

**Review by kolodny — CHANGES_REQUESTED** [review]
Awesome! Thanks so much for this!

**Comment by kolodny** [comment]
Thanks! published to `v2.6.0`

**Pull request merged**